addons/l10n_br/models/account_move.py

```diff
@@ -0,0 +1,39 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import models, fields
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    l10n_br_invoice_serial = fields.Char(related="journal_id.l10n_br_invoice_serial", readonly=True)
+
+    def _compute_l10n_latam_document_type(self):
+        """ Override for debit notes. This sets the same document type as the one on the origin. Cannot
+         override the defaults in the account.move.debit wizard because l10n_latam_invoice_document explicitly
+         calls _compute_l10n_latam_document_type() after the debit note is created. """
+        br_debit_notes = self.filtered(lambda m: m.state == "draft" and m.country_code == "BR" and m.debit_origin_id.l10n_latam_document_type_id)
+        for move in br_debit_notes:
+            move.l10n_latam_document_type_id = move.debit_origin_id.l10n_latam_document_type_id
+
+        return super(AccountMove, self - br_debit_notes)._compute_l10n_latam_document_type()
+
+    def _get_last_sequence_domain(self, relaxed=False):
+        """ Override to give sequence names in the same journal their own, independent numbering. """
+        where_string, param = super()._get_last_sequence_domain(relaxed)
+        if self.company_id.account_fiscal_country_id.code == "BR" and self.l10n_latam_use_documents:
+            where_string += " AND l10n_latam_document_type_id = %(l10n_latam_document_type_id)s "
+            param["l10n_latam_document_type_id"] = self.l10n_latam_document_type_id.id or 0
+        return where_string, param
+
+    def _get_l10n_latam_documents_domain(self):
+        """ Override to support an empty internal_type. These can be used on all documents. """
+        self.ensure_one()
+        if self.country_code == "BR":
+            internal_types = [False, ""]
+            if self.move_type in ["out_refund", "in_refund"]:
+                internal_types.append("credit_note")
+            else:
+                internal_types.extend(["invoice", "debit_note"])
+            return [("internal_type", "in", internal_types), ("country_id", "=", self.company_id.account_fiscal_country_id.id)]
```

Comment:
I wonder, is there a particular reason why this was preferable to first getting a domain using `super()` and then extending it with `[False, ""]`?

Like in `l10n_ar`: https://github.com/odoo/odoo/blob/46f1e7552cd99025325cc69d9715ac1bf48c0059/addons/l10n_ar/models/account_move.py#L116

Reply:
Hey @antoine162 thanks for the review!
About this point:

> I was wondering whether it would be useful to put some demo data in the l10n_br_invoice_serial field of the journals for the demo company?
I agree, it will be nice - on demo data - to add a serial = 1 on the Sales Journal only, on the Vendor Side - purchase journal - we can leave it empty.

Thanks!
cc' @jorenvo

Reply:
Ahhhh :smile: I understand, thanks for the explanation - I hadn't thought about all this!
Yeah, to me your existing code (without the `super()`) does feel 'cleaner'. :+1:

Reply:
No problem! If everything looks ok to you, can you r+? I don't have r+ rights myself.

Reply:
I don't either :smile: , it's for @jco-odoo

Reply:
See on the second commit of https://github.com/odoo/odoo/pull/136413.  I think we can make it generic that if the internal type is empty, it always appears.